### PR TITLE
mgr/dashboard: Fix search in `Source` column of RBD configuration list

### PR DESCRIPTION
**Pull request by p-na**
Fixes: http://tracker.ceph.com/issues/37569

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts [hunk 1]
```diff
@@ -509,20 +510,22 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
...
-          let cellValue: any = _.get(d, c.prop);
-          if (!_.isUndefined(c.pipe)) {
-            cellValue = c.pipe.transform(cellValue);
+        columns.filter((col) => {
+          let cellValue: any = _.get(row, col.prop);
+
+          if (!_.isUndefined(col.pipe)) {
+            cellValue = col.pipe.transform(cellValue);
           }
           if (_.isUndefined(cellValue)) {
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts]: The UI crashes below when ``cellValue`` is ``null``. Shouldn't we exit here immediately?.

auswahl_002

Comment (p-na) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts]: I'm not sure how to reproduce it, can you please explain how to do it?

And, am I correctly assuming that you are referring to the implementation already in master, as I've only refactored variable names in this code, but not the functionality?

Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts]: I've an up-to-date master and pulled your PR. After that i created a replicated pool and two RBD's as you can see in the upper screenshot.

auswahl_003

Finally i only typed a ``t`` in the search field.

Comment (p-na) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts]: You found a bug. I created a separate [issue](http://tracker.ceph.com/issues/38583) for this as this problem has been reproducible on master. There's also a [pull request](https://github.com/ceph/ceph/pull/26766) with your suggested solution.

**Pull request merged**